### PR TITLE
Fix/adf 519/reindexation while moving classes

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-519

deployed at http://adf-519-move-class-with-search.playground.kitchen.it.taocloud.org:47351/ for QA and curious

### CODE REVIEW

File: models/classes/GenerisServiceTrait.php [hunk 1]
```diff
@@ -308,7 +308,11 @@ public function changeClass(core_kernel_classes_Resource $instance, core_kernel_
         if ($instance->isClass()) {
             try {
                 /** @var core_kernel_classes_Class $instance */
-                return $instance->editPropertyValues($this->getProperty(OntologyRdfs::RDFS_SUBCLASSOF), $destinationClass);
+                $status =  $instance->editPropertyValues($this->getProperty(OntologyRdfs::RDFS_SUBCLASSOF), $destinationClass);
+                if ($status){
+                    $this->getEventManager()->trigger(new ClassMovedEvent($instance));
```
Comment (gabrielfs7) [inline models/classes/GenerisServiceTrait.php:313]: Did not we already have an Event objet for this already (that is listened on AdvancedSearch indexatio)?

File: models/classes/search/tasks/UpdateClassInIndex.php [hunk 2]
```diff
@@ -50,19 +51,17 @@ public function __construct(IndexIteratorFactory $indexIteratorFactory = null)
...
      */
     public function __invoke($params): Report
     {
         if (empty($params) || empty($params[0])) {
             throw new common_exception_MissingParameter();
         }
 
-        $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
+        /** @var SearchProxy $searchService */
+        $searchService = $this->getServiceLocator()->get(SearchProxy::class);
```
Comment (gabrielfs7) [inline models/classes/search/tasks/UpdateClassInIndex.php:64]: We need to stick to SERVICE_ID

```suggestion
        $searchService = $this->getServiceLocator()->get(SearchProxy::SERVICE_ID);
```

File: models/classes/search/index/IndexIterator.php [hunk 3]
```diff
@@ -15,33 +15,26 @@
...
 
-/**
- * Class IndexIterator
- * @package oat\tao\model\search\index
- */
 class IndexIterator extends \IteratorIterator implements ServiceLocatorAwareInterface
 {
     use ServiceLocatorAwareTrait;
 
     /** @var IndexService  */
```
Comment (gabrielfs7) [inline models/classes/search/index/IndexIterator.php:31]:
```suggestion
    /** @var IndexDocumentBuilder  */
```

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/GenerisServiceTrait.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/GenerisServiceTrait.php:313]: Will this be triggered for all the subclasses too?

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/GenerisServiceTrait.php [hunk 4]
```diff
@@ -308,7 +308,11 @@ public function changeClass(core_kernel_classes_Resource $instance, core_kernel_
         if ($instance->isClass()) {
             try {
                 /** @var core_kernel_classes_Class $instance */
-                return $instance->editPropertyValues($this->getProperty(OntologyRdfs::RDFS_SUBCLASSOF), $destinationClass);
+                $status =  $instance->editPropertyValues($this->getProperty(OntologyRdfs::RDFS_SUBCLASSOF), $destinationClass);
```
Comment (shpran) [inline models/classes/GenerisServiceTrait.php:311]:
```suggestion
                $status = $instance->editPropertyValues($this->getProperty(OntologyRdfs::RDFS_SUBCLASSOF), $destinationClass);
```

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/GenerisServiceTrait.php (hunk 1 above)
Comment (boajer) [reply models/classes/GenerisServiceTrait.php:313]: We have only one event triggered, but within listener it's handled in the way to take care about all subclasses too https://github.com/oat-sa/extension-tao-advanced-search/pull/39/files#diff-f973ef466a2b9ce3babcae11f083701e1adcbab256f53c49eabd597bf6291872R45-R49

**Pull request merged**